src/fmpz/CRT.c

```diff
@@ -65,16 +65,14 @@ _fmpz_CRT(fmpz_t out, const fmpz_t r1, const fmpz_t m1, fmpz_t r2,
 }
 
 void fmpz_CRT(fmpz_t out, const fmpz_t r1, const fmpz_t m1,
-    fmpz_t r2, fmpz_t m2, int sign)
+    const fmpz_t r2, const fmpz_t m2, int sign)
 {
     fmpz_t m1m2, c;
 
     fmpz_init(c);
 
     fmpz_mod(c, m1, m2);
-    fmpz_invmod(c, c, m2);
-
-    if (fmpz_is_zero(c))
+    if (!fmpz_invmod(c, c, m2))
```

Comment:
`fmpz_invmod(f,g,h)` returns 0 if `g` is not invertible modulo `h`. It doesn't necessarily set `f=0`.